app/models/service_template.rb

```diff
@@ -391,9 +391,45 @@ def self.create_from_options(options)
   private_class_method :create_from_options
 
   def provision_request(user, options = nil, request_options = nil)
-    result = provision_workflow(user, options, request_options).submit_request
-    raise result[:errors].join(", ") if result[:errors].any?
-    result[:request]
+    result = order(user, options, request_options)
+    result[:request] || raise(result[:errors].join(", "))
```

Comment:
Good point @carbonin, this also concerns me.

Reply:
After a look it does look like the result will only ever have a request if there are no errors, but because this change didn't have anything to do with the intention of the PR I'm going to move it back to the way it was.